### PR TITLE
Raul/redux logic/cmji 59

### PR DESCRIPTION
**Pull request by raulmogos**

### CODE REVIEW

File: src/actions/index.js [hunk 1]
```diff
@@ -7,3 +18,41 @@ export const fetchContacts = () => ({
...
+
+export const deleteContact = id => ({
+  type: DELETE_CONTACT,
+  payload: id
+});
+
+export const addContact = formValues => ({
+  type: ADD_CONTACT,
+  payload: {
+    ...formValues,
```
Comment (raresfelecan) [inline src/actions/index.js]: `actions` should not have any logic. Only send `formValues` as payload and create the new object with the required fields in the reducer.

File: src/components/contact/contact.container.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import {
+  changeIsChecked,
+  deleteContact,
+  increaseLikes,
+  decreaseLikes,
+  fetchFavourites
+} from '../../actions';
+
+const mapStateToProps = (state, ownProps) => ({
+  contact: state.contactsData.contacts.find(x => x.id === ownProps.id)
```
Comment (raresfelecan) [inline src/components/contact/contact.container.js]: This is not good! This parses the array for each contact. Pretty hard performance hit. In this case you can just leave it as it was before, `foreach(contact in contacts) -> render component`

Comment (raresfelecan) [reply src/components/contact/contact.container.js]: And send the `contact` with props from contacts-list

File: src/components/contacts-list/contacts-list.component.js [hunk 3]
```diff
@@ -20,7 +20,7 @@ class ContactsList extends React.Component {
     return contactsList.map(item => (
       <Contact
         key={item.id}
-        contact={item}
```
Comment (raresfelecan) [inline src/components/contacts-list/contacts-list.component.js]: Here! Revert this as it is the correct way!

File: src/components/contacts-page/contacts-page.component.js [hunk 4]
```diff
@@ -1,114 +1,53 @@
...
+    _clearContacts();
   }
 
   anyContactSelected = () => {
-    const { contacts } = this.state;
+    const { contacts } = this.props;
     return contacts.some(x => x.isChecked);
   }
 
   deleteSelectedContacts = () => {
```
Comment (raresfelecan) [inline src/components/contacts-page/contacts-page.component.js]: Now this function only calls another function. Remove this and use the one from props directly where you need it

File: src/components/contacts-page/contacts-page.component.js [hunk 5]
```diff
@@ -1,114 +1,53 @@
...
-    this.setState({ contacts: [newContact, ...contacts] });
-  }
 
   areContactsWithLikes = () => {
-    const { contacts } = this.state;
+    const { contacts } = this.props;
     return contacts.some(item => item.likes);
   };
 
   clearAllContactsLikes = () => {
```
Comment (raresfelecan) [inline src/components/contacts-page/contacts-page.component.js]: Now this function only calls another function. Remove this and use the one from props directly where you need it

File: src/components/contact/contact.component.js [hunk 6]
```diff
@@ -1,40 +1,56 @@
...
-    const { contactMethods, contact } = this.props;
-    contactMethods.changeIsChecked(contact.id);
+    const { _changeIsChecked, contact } = this.props;
+    _changeIsChecked(contact.id);
   }
 
   increaseLikes = () => {
-    const { contactMethods, contact } = this.props;
+    const { _increaseLikes, contact } = this.props;
     if (contact.likes >= LIKES.MAX) return;
```
Comment (raresfelecan) [inline src/components/contact/contact.component.js]: I think here you can do it like this:
```(contact.likes < LIKES.MAX) && _increaseLikes(contact.id);```
and the same for `decreaseLikes`

File: src/components/contact/contact.component.js [hunk 7]
```diff
@@ -1,40 +1,56 @@
...
-    contactMethods.updateLikes(contact.id, 1);
+    _increaseLikes(contact.id);
   }
 
   decreaseLikes = () => {
-    const { contactMethods, contact } = this.props;
+    const { _decreaseLikes, contact } = this.props;
     if (contact.likes <= LIKES.MIN) return;
-    contactMethods.updateLikes(contact.id, -1);
+    _decreaseLikes(contact.id, -1);
```
Comment (raulmogos) [inline src/components/contact/contact.component.js]: the second argument (-1) will be removed

**Pull request merged**